src/prefect/tasks/prefect/flow_run.py

```diff
@@ -224,16 +224,16 @@ def wait_for_flow_run(
 
     flow_run = FlowRunView.from_flow_run_id(flow_run_id)
 
-    def log_with_flow_run_prefix(log: FlowRunLog):
-        message = f"Flow {flow_run.name!r}: {log.message}"
-        prefect.context.logger.log(log.level, message)
-
-    output_fn = log_with_flow_run_prefix if stream_state else lambda *_, **__: None
-
     if not stream_state and stream_logs:
         warnings.warn("`stream_logs` will be ignored since `stream_state` is `False`")
 
-    return watch_flow_run(flow_run_id, stream_logs=stream_logs, output_fn=output_fn)
+    for log in watch_flow_run(flow_run_id, stream_logs=stream_logs):
+        if stream_state:
```

Comment:
is it worth passings `stream_state` as an arg to `watch_flow_run`? feels confusing to effectively use `stream_state` and `stream_logs` to restrict logs in different interfaces 

`wait_for_flow_run` is responsible for handling `stream_state` while `watch_flow_run` is PARTIALLY responsible for handling `stream_logs` because `wait_for_flow_run.stream_state` will affect the output

Reply:
I guess my presumption was `watch_flow_run` will always stream states because otherwise it's silly to watch it but I could be convinced to make state streaming toggle there as well.

Reply:
I agree this feels a little clunky as is. I could also just change the names of the variables?